controller/token_storage_blackbox_test.go

```diff
@@ -305,6 +305,127 @@ func (rest *TestTokenStorageREST) TestRetrieveExternalTokenValidOnForcePullInter
 
 }
 
+// Not present in DB but present in Keycloak
+func (rest *TestTokenStorageREST) TestStatusExternalTokenGithubOK() {
+	identity, err := testsupport.CreateTestIdentity(rest.DB, uuid.NewV4().String(), "KC")
+	require.Nil(rest.T(), err)
+	rest.mockKeycloakExternalTokenServiceClient.scenario = "positive"
+	service, controller := rest.SecuredControllerWithIdentityAndDummyProviderFactory(identity)
+	test.StatusTokenOK(rest.T(), service.Context, service, controller, "https://github.com/a/b", nil)
+	test.StatusTokenOK(rest.T(), service.Context, service, controller, "https://api.starter-us-east-2.openshift.com", nil)
+}
+
+// Not present in Keycloak but present in DB.
+func (rest *TestTokenStorageREST) TestStatusExternalTokenPresentInDB() {
+	rest.statusExternalTokenFromDBSuccess("unlinked")
+}
+
+// Get token from keycloak fails for any reason but token present in DB.
+func (rest *TestTokenStorageREST) TestStatusExternalTokenFailedInKeycloak() {
+	rest.statusExternalTokenFromDBSuccess("internalError")
+}
+
+func (rest *TestTokenStorageREST) statusExternalTokenFromDBSuccess(scenario string) account.Identity {
+	identity, _ := rest.retrieveExternalTokenFromDBSuccess(scenario)
+	service, controller := rest.SecuredControllerWithIdentityAndDummyProviderFactory(identity)
+	test.StatusTokenOK(rest.T(), service.Context, service, controller, "https://github.com/a/b", nil)
+	return identity
+}
+
+// Not present in DB and failed in Keycloak for any reason
+func (rest *TestTokenStorageREST) TestStatusExternalTokenUnauthorized() {
+	rest.checkStatusExternalTokenUnauthorized("https://github.com/sbose78", "github", "unlinked")
+	rest.checkStatusExternalTokenUnauthorized("https://api.starter-us-east-2.openshift.com", "openshift-v3", "internalError")
+	rest.checkStatusExternalTokenUnauthorized("https://github.com/sbose78", "github", "unlinked")
+	rest.checkStatusExternalTokenUnauthorized("https://api.starter-us-east-2.openshift.com", "openshift-v3", "internalError")
+}
+
+func (rest *TestTokenStorageREST) checkStatusExternalTokenUnauthorized(for_ string, providerName string, kcScenario string) {
+	identity, err := testsupport.CreateTestIdentity(rest.DB, uuid.NewV4().String(), "KC")
+	require.Nil(rest.T(), err)
+	rest.mockKeycloakExternalTokenServiceClient.scenario = kcScenario
+
+	service, controller := rest.SecuredControllerWithIdentity(identity)
+
+	rw := httptest.NewRecorder()
+	u := &url.URL{
+		Scheme: "https",
+		Host:   "auth.localhost.io",
+		Path:   fmt.Sprintf("/api/token"),
```

Comment:
response would be the same, but still we should have a test for `/api/token/status` .

Reply:
Good catch! Fixed in - https://github.com/fabric8-services/fabric8-auth/pull/332/commits/f79f03d89f2fa4577ae5dd1c3d8cd933fe5f82ed